Edit Privacy 2021 Chapter

tunetheweb opened the pull request:
Closes #2149 

A bit of a big edit I'm afraid, primarily due to:
- Move of figures to before the discussion (in the Web Almanac we prefer to present the data and then discuss it - since merged in #2477 ).
- Breaking up of longer paragraphs into two (since merged in #2477 ).
- Switch to percentage from absolute values (I left absolute for some of the vey small numbers).
- This required changing and rerunning some of the queries.
- Consistency of decimal places (have used two decimal places throughout this chapter).
- Addition of much more detail figure descriptions.

There's a few TODOs left to action or remove. Will highlight them.

This chapter has been staged here: https://20211112t221945-dot-webalmanac.uk.r.appspot.com/en/2021/privacy

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
-
+Online tracking is often done through third-party libraries. These libraries usually provide some (useful) service, but in the process some of them also generate a unique identifier for each user, which can then be used to follow and profile users across websites. The <a hreflang="en" href="https://whotracks.me/">WhoTracksMe</a> project is dedicated to discovering the most widely deployed online  ...
 
+{# TODO - Can you explain _why_ you are restricting yourselves to just those four categories? #}
```
Suggestion to add an explanation here as to _why_ you are restricting yourselves to just those four categories?

VictorLeP replied:
see d87c1a4

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
   )
 }}
 
+{# TODO - this SQL doesn't exist. Also why only desktop? #}
```
We are missing the CNAME SQL queries, meaning I couldn't convert to percentage

VictorLeP replied:
@ydimova

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
   )
 }}
 
+{# TODO - this SQL doesn't exist. Also why only desktop? #}
```
We are missing the CNAME SQL queries, meaning I couldn't convert to percentage

VictorLeP replied:
@ydimova

ydimova replied:
Added the CNAME SQL queries to the branch

VictorLeP replied:
Fixed the linting for these queries as well.

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
   )
 }}
 
+{# TODO - this SQL doesn't exist. Also why desktop? #}
```
We are missing the CNAME SQL queries, meaning I couldn't convert to percentage

VictorLeP replied:
@ydimova

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
-Some websites request access to specific features and browser APIs that can impact the user's privacy, for instance by accessing the geolocation data, microphone, camera, etc. While these features usually serve very useful purposes, such as discovering nearby points of interest or allowing people to communicate with each other, there is a risk of exposing sensitive data if the user does not fully ...
+Some websites request access to specific features and browser APIs that can impact the user's privacy, for instance by accessing the geolocation data, microphone, camera, etc. While these features usually serve very useful purposes, such as discovering nearby points of interest or allowing people to communicate with each other, there is a risk of exposing sensitive data if the user does not fully ...
 
+{# TODO I feel we should make it clearer that these cannot be activated by user consent (though of course once that consent's been given, it can be abused!). Otherwise is comes across as a bit inaccurate and fear mongering #}
 
```
Suggestion: I feel we should make it clearer that these cannot be activated by user consent (though of course once that consent's been given, it can be abused!). Otherwise is comes across as a bit inaccurate and fear mongering

VictorLeP replied:
see d87c1a4

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
   )
 }}
 
+{# TODO - do we have anything to say about this chart? Perhaps that it's a year-round problem and that the recent tail off is probably more indiciative of under reporting as some breaches will not have been discovered and/or made public yet? If we don't have anything to say then should we just drop the chart? #}
```
Do we have anything to say about this chart? Perhaps that it's a year-round problem and that the recent tail off is probably more indiciative of under reporting as some breaches will not have been discovered and/or made public yet? If we don't have anything to say then should we just drop the chart?

VictorLeP replied:
I would be fine with dropping the chart, but adding a few more details to the text (a.o. the number of sensitive breaches).

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
   )
 }}
 
+{# TODO - are email addresses _always_ leaked? Or only in HaveIBeenPwned's dataset since that is the primary key they use? #}
```
Question for clarification

VictorLeP replied:
There are a handful breaches that are keyed on phone numbers, but it's mostly email addresses. Breaches without either would indeed probably not be loaded into HIBP since there's no way to search for your records.

tunetheweb replied:
Then I think we should either drop this sentence, or change it to:

> The email addresses of users are leaked in ever breach that HaveIPeenPwned tracks since that is what it uses to allow users to query it's service.

Or something like that. That is drop emphasis on _every_ and explain more.

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
-The most common referrer policy that is explicitly set is `no-referrer-when-downgrade`. It's set on 3.38% of websites on mobile clients and 3.81% of websites on desktop clients. `no-referrer-when-downgrade` is not privacy-enhancing. With this policy, full URLs of pages a user visits on a given site are shared in cross-origin requests, which makes this information accessible to other parties (orig ...
-
-In addition, around 0.5% of websites set the value of the referrer policy to `unsafe-url`, which allows the origin, host and query string to be sent with _any_ request, regardless of the security level of the receiver. In this case, a referrer could be sent in the clear, potentially leaking private information. Worryingly, sites are actively being configured to enable this behavior.
+{# TODO Worth pointing out that this tighting up of what is shown is fairly recent? Not sure if there's a reference for that but seem to recall it was in last year or two. #}
```
Suggestion.

VictorLeP replied:
see d87c1a4

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
-Below, we show the distribution of the usage of both versions of TCF and of USP. Note that the crawl is US-based, therefore we do not expect many websites to have implemented TCF. Fewer than 2% of websites use any TCF version, while twice as many websites use the US Privacy framework.
+The <a hreflang="en" href="https://iabeurope.eu/transparency-consent-framework/">Transparency and Consent Framework</a> (TCF) is an initiative of the Interactive Advertising Bureau Europe (IAB) for providing an industry standard for communicating user consent to advertisers. The framework consists of a <a hreflang="en" href="https://iabeurope.eu/vendor-list/">Global Vendor List</a>, in which vend ...
 
+TCF is meant as a GDPR-compliant mechanism in Europe, although [a recent decision by the Belgian Data Protection Authority](https://iabeurope.eu/all-news/update-on-the-belgian-data-protection-authoritys-investigation-of-iab-europe/){# TODO decide if using that URL or https://www.iccl.ie/news/online-consent-pop-ups-used-by-google-and-other-tech-firms-declared-illegal/ #} found that this system is  ...
```
This was set to two URLs with a TODO to decide, but looks like you forgot to decide. I picked the first and put the other in a comment. Can you review?

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
 
+TCF is meant as a GDPR-compliant mechanism in Europe, although [a recent decision by the Belgian Data Protection Authority](https://iabeurope.eu/all-news/update-on-the-belgian-data-protection-authoritys-investigation-of-iab-europe/){# TODO decide if using that URL or https://www.iccl.ie/news/online-consent-pop-ups-used-by-google-and-other-tech-firms-declared-illegal/ #} found that this system is  ...
 
+{# TODO what's the USP? Why not CFPTC? #}
```
Is this acronym correct? If so, should it be explained.

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
+In the 10 most popular consent management platforms that are part of the framework, at the top we find <a hreflang="en" href="https://www.quantcast.com/products/choice-consent-management-platform/">Quantcast</a> with 0.34% on mobile. Other popular solutions are <a hreflang="en" href="https://www.didomi.io/">Didomi</a> with 0.24%, and Wikia, with 0.30%.
 
-In the USP framework, the website's and user's privacy settings are encoded in a "<a hreflang="en" href="https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/US%20Privacy%20String.md">privacy string</a>". The most common privacy string is `1---`. This indicates that CCPA does not apply to the website and therefore the website not obliged to provide an opt-out for the user. C ...
+{# TODO why no link for Wikia? #}
```
Question

VictorLeP replied:
Wikia appears to have a custom CMP, but doesn't provide it to other websites. It just happens to have a lot of sites in CRuX probably, which makes it one of the biggest "CMP providers". I didn't add a link because there's basically nothing to see there (about CMPs).

tunetheweb replied:
Just feels a bit weird that it's the only one not linked. But if don't have a relevant link then not force on in for the sake of it. Resolving.

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,541 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
+Around the same percentage of pages on mobile and desktop clients use DNT. However, in practice hardly any websites actually respect the DNT opt-outs. The Tracking Protection Working Group, which specifies DNT, <a hreflang="en" href="https://www.w3.org/2016/11/tracking-protection-wg.html">closed down</a> in 2018, due to <a hreflang="en" href="https://lists.w3.org/Archives/Public/public-tracking/2 ...
 
-DNT's successor <a hreflang="en" href="https://globalprivacycontrol.org/">Global Privacy Control</a> (GPC) was released in October 2020 and is meant to provide a more enforceable alternative, with the hopes of better adoption. This privacy preference signal is implemented with a single bit in all HTTP requests. We did not yet observe any uptake, but we can expect this to improve in future as <a h ...
+{# TODO - worth pointing out WHY Safari removed it? "to prevent potential use as a fingerprinting variable." #}
```
Suggestion

VictorLeP commented:
@maudnals I've added you as a reviewer, since I'm seeing a few changes similar to ones that you disagreed with while drafting the text, but feel free to decline :)

tunetheweb commented:
All, I've pulled some of the paragraph splitting and figures moves into a separate PR: https://github.com/HTTPArchive/almanac.httparchive.org/pull/2477 and merged that. Hoping that should make this PR, with the "real edits", easier to review.

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -212,76 +231,78 @@ Sensors can be useful to make a website more interactive, but could also be abus
 
-Geolocations services provide GPS and other location data of the user (such as <a hreflang="en" href="https://developer.mozilla.org/en-US/docs/Glossary/IP_Address">IP address</a>) and can be used by trackers to provide more relevant content to the user among other things.
-Therefore, we analyze the use of geolocation-as-a-service technologies on websites, based on libraries detected through [Wappalyzer](./methodology#wappalyzer).
+Geolocations services provide GPS and other location data of the user (such as [IP address](https://developer.mozilla.org/en-US/docs/Glossary/IP_Address)) and can be used by trackers to provide more relevant content to the user among other things. Therefore, we analyze the use of "geolocation-as-a-service" technologies on websites, based on libraries detected through [Wappalyzer](./methodology#wa ...
```
```suggestion
Geolocation services provide GPS and other location data (such as [IP address](https://developer.mozilla.org/en-US/docs/Glossary/IP_Address)) of the user and can be used by trackers to provide more relevant content to the user among other things. Therefore, we analyze the use of "geolocation-as-a-service" technologies on websites, based on libraries detected through [Wappalyzer](./methodology#wappalyzer).
```

tunetheweb replied:
Happy for you to directly edit this branch and make these minor changes of things you've spotted in addition.

If reverting any of my suggested changes however could you add as comment instead to discuss?

VictorLeP replied:
I didn't notice that I reverted a change! You moved the 'of the user' then?

tunetheweb replied:
You didn't. I meant if you were going to.

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -212,76 +231,78 @@ Sensors can be useful to make a website more interactive, but could also be abus
   image="data_breaches_pwned_websites_per_date_and_breach_sensitivity.png",
   caption="Data breaches in the previous year and their sensitivity",
-  description="Chart that shows the number of domains involved in a breach over the course of the previous year and the sentitivity of the data breach.",
+  description="Chart that shows the number of domains involved in a breach over the course of the previous year and the sensitivity of the data breach. For Not Sensitive breaches it was 4 in August 2020, 2 in September, 6 in October, 1 in November, 6 in December, 5 in January 2021, 3 in February, 5 in March, 2 in April 1 in May, 1 in June, and finally 1 in July. For Sensitive breaches it's just 1 ...
```
```suggestion
  description="Chart that shows the number of domains involved in a breach over the course of the previous year and the sensitivity of the data breach. For Not Sensitive breaches it was 4 in August 2020, 2 in September, 6 in October, 1 in November, 6 in December, 5 in January 2021, 3 in February, 5 in March, 2 in April, 1 in May, 1 in June, and finally 1 in July. For Sensitive breaches it's just 1 in February 2021 and 2 in March 2021.",
```

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -290,111 +311,111 @@ While you're browsing the web, there is certain data that you might want to keep
   image="most_common_featurepolicy_permissionspolicy_directive_values.png",
   caption="Values used for the most popular feature policy directives",
-  description="Bar chart showing the distribution of the values assigned to the 3 most popular directives for the feature policy.",
+  description="Bar chart showing the distribution of the values assigned to the 3 most popular directives for the feature policy. `microphone` is set to `self` for 0.08% of mobiles sites, it is set to `none` for 0.49% of sites, and is set to `*` for 0.03% of sites. `geolocation` is set to `self` for 0.17% of mobiles sites, it is set to `none` for 0.34% of sites, and is set to `*` for 0.05% of sit ...
```
```suggestion
  description="Bar chart showing the distribution of the values assigned to the 3 most popular directives for the feature policy. `microphone` is set to `self` for 0.08% of mobile sites, it is set to `none` for 0.49% of sites, and is set to `*` for 0.03% of sites. `geolocation` is set to `self` for 0.17% of mobile sites, it is set to `none` for 0.34% of sites, and is set to `*` for 0.05% of sites. `camera` is set to `self` for 0.09% of mobile sites, it is set to `none` for 0.46% of sites, and is set to `*` for 0.03% of sites.",
```

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -290,111 +311,111 @@ While you're browsing the web, there is certain data that you might want to keep
 ## How websites give you a privacy choice: Privacy preference signals
 
-In light of the recent introduction of privacy regulations, such as the <a hreflang="en" href="https://ec.europa.eu/info/law/law-topic/data-protection/data-protection-eu">GDPR</a> in Europe, <a hreflang="en" href="https://www.oag.ca.gov/privacy/ccpa">CCPA</a> in California, <a hreflang="pt-br" href="https://www.gov.br/cidadania/pt-br/acesso-a-informacao/lgpd">LGPD</a> in Brazil, the <a hreflang=" ...
+In light of the recent introduction of privacy regulations, such as those mentioned in the introduction and more, websites are required to obtain explicit user consent about the collection of personal data for any non-essential features such as marketing and analytics.
```
```suggestion
In light of the recent introduction of privacy regulations, such as those mentioned in the introduction, websites are required to obtain explicit user consent about the collection of personal data for any non-essential features such as marketing and analytics.
```
Double indication that there are others (such as/and more)

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -85,44 +90,46 @@ The main technical approach to store and retrieve cross-site user identifiers is
-  height=421,
-  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRfOwcNkLp-mYBkmhDV5AFxl8p0ls9oxFnmmo0WUcAJxjxQqmHjquRZCWj_lNZRyFtX3RdH5T92IESu/pubchart?oid=932933299&format=interactive",
+  caption="Top 10 domains setting cookies from header.",
+  description="Chart showing the percentage of websites that include a cookie set via the response header for the 10 most popular tracking domains setting cookies. `doubleclick.net` is used on 30.49% of desktop pages and 28.72% of mobile pages, `facebook.com` on 23.075 and 21.43% respectively, `youtube.com` on 10.02% and 8.83%, `google.com` on 8.62% and 8.45%, `yandex.ru` on 4.42% and 5.17%, `pub ...
```
```suggestion
  description="Chart showing the percentage of websites that include a cookie set via the response header for the 10 most popular tracking domains setting cookies. `doubleclick.net` is used on 30.49% of desktop pages and 28.72% of mobile pages, `facebook.com` on 23.07 and 21.43% respectively, `youtube.com` on 10.02% and 8.83%, `google.com` on 8.62% and 8.45%, `yandex.ru` on 4.42% and 5.17%, `pubmatic.com` on 3.82% and 4.73%, `rlcdn.com` on 4.01% and 3.99%, `openx.net` on 3.57% and 4.42%, `adsrvr.org` on 4.00% and 3.90%, and finally `yahoo.com` on 3.80% and 3.70%.",
```

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -133,77 +140,89 @@ From the percentage of websites using these third-party services, we can see tha
 Some websites request access to specific features and browser APIs that can impact the user's privacy, for instance by accessing the geolocation data, microphone, camera, etc. While these features usually serve very useful purposes, such as discovering nearby points of interest or allowing people to communicate with each other, there is a risk of exposing sensitive data if the user does not fully ...
 
-We'll look at how often websites request access to sensitive resources. Moreover, any time a service stores sensitive data, there is the danger of hackers stealing and leaking that data. We'll look at recent data breaches that prove that this danger is real.
+{# TODO I feel we should make it clearer that these cannot be activated by user consent (though of course once that consent's been given, it can be abused!). Otherwise is comes across as a bit inaccurate and fear mongering #}
```
Remaining question

tunetheweb replied:
Happy to resolve this base don your changes if that's a remaining question for me? Also happy to leave open if you're still thinking about it and this is a pointer for you?

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -133,77 +140,89 @@ From the percentage of websites using these third-party services, we can see tha
   image="nb_websites_with_device_sensor_events.png",
   caption="5 most used sensor events",
-  description="Bar chart showing the most widely accessed sensor events, based on the use of JavaScript listeners.",
+  description="Bar chart showing the most widely accessed sensor events, based on the use of JavaScript listeners. `deviceOrientation` is found on 3.32% of desktop sites and 3.23% of mobile, `deviceReady` on 1.12% and 1.23%, `devicemotion` on 0.65% and 0.66%, `deviceChange` on 0.03% and 0.02%, and finally `deviceproximity` on 0.03% and 0.02%.",
```
```suggestion
  description="Bar chart showing the most widely accessed sensor events, based on the use of JavaScript listeners. `deviceOrientation` is found on 3.32% of desktop sites and 3.23% of mobile sites, `deviceReady` on 1.12% and 1.23%, `devicemotion` on 0.65% and 0.66%, `deviceChange` on 0.03% and 0.02%, and finally `deviceproximity` on 0.03% and 0.02%.",
```

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -212,76 +231,78 @@ Sensors can be useful to make a website more interactive, but could also be abus
 }}
 
-We find that the most popular service, <a hreflang="en" href="https://www.ipify.org/">ipify</a>, is used on 0.09% of desktop websites and 0.07% of mobile websites.
+We find that the most popular service, <a hreflang="en" href="https://www.ipify.org/">ipify</a>, is used on 0.09% of desktop websites and 0.07% of mobile websites. So, it would appear that few websites utilize geolocation services.
```
```suggestion
We find that the most popular service, <a hreflang="en" href="https://www.ipify.org/">ipify</a>, is used on 0.09% of desktop websites and 0.07% of mobile websites. So, it would appear that few websites use geolocation services.
```

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -290,111 +311,111 @@ While you're browsing the web, there is certain data that you might want to keep
-  description="Bar chart showing the most common directives used to define the feature policy and the number of websites that are using them.",
-  height=421,
-  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRfOwcNkLp-mYBkmhDV5AFxl8p0ls9oxFnmmo0WUcAJxjxQqmHjquRZCWj_lNZRyFtX3RdH5T92IESu/pubchart?oid=1180513630&format=interactive",
+  description="Bar chart showing the most common directives used to define the feature policy and the number of websites that are using them. `geolocation` is used by 2,222 of desktop sites and 2,323 of mobile sites, `microphone` by 2,199 and 2,310 respectively, `camera` by 2,082 and 2,197, `payment` by 1,748 and 1,879, `usb` by 1,354 and 1,492, `gyroscope` by 1,145 and 1,025, `magnetometer` by 1 ...
```
```suggestion
  description="Bar chart showing the most common directives used to define the feature policy and the number of websites that are using them. `geolocation` is used by 2,222 desktop sites and 2,323 mobile sites, `microphone` by 2,199 and 2,310 respectively, `camera` by 2,082 and 2,197, `payment` by 1,748 and 1,879, `usb` by 1,354 and 1,492, `gyroscope` by 1,145 and 1,025, `magnetometer` by 1,141 and 1,024, `interest-cohort` by 1,037 and 1,019, `fullscreen` by 940 and 873, `accelerometer` by 892 and 852.",
```

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -290,111 +311,111 @@ While you're browsing the web, there is certain data that you might want to keep
-  caption="Percentage of websites that use a consent management platform",
-  description="Bar chart showing the percentage of websites that use a third-party library for consent management.",
+  caption="Percentage of websites that use a Consent Management Platform",
+  description="Bar chart showing the percentage of websites that use a third-party library for consent management. 7.10% of desktop sites and 6.97% of mobile sites us a CMP.",
```
```suggestion
  description="Bar chart showing the percentage of websites that use a third-party library for consent management. 7.10% of desktop sites and 6.97% of mobile sites use a CMP.",
```

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -437,25 +462,28 @@ Above, we show the distribution of the usage of both versions of TCF and of USP.
   caption="Percentage of websites using IAB US privacy strings",
-  description="Bar chart showing the percentage of websites that use each privacy string for IAB US' consent framework.",
-  height=421,
+  description="Bar chart showing the percentage of websites that use each privacy string for IAB US' consent framework. `1---` is used by 0.87% of desktop websites and 0.80% of mobile websites, `1YNY` is 0.72% and 0.64% respectively, `1YNN` is 0.07% and 0.06%, and blank is 0.01% and 0.00%.",
```
```suggestion
  description="Bar chart showing the percentage of websites that use each privacy string for IAB's USP consent framework. `1---` is used by 0.87% of desktop websites and 0.80% of mobile websites, `1YNY` is 0.72% and 0.64% respectively, `1YNN` is 0.07% and 0.06%, and blank is 0.01% and 0.00%.",
```

VictorLeP commented on src/content/en/2021/privacy.md:
```diff
@@ -466,25 +494,24 @@ The most common privacy string is `1---`. This indicates that CCPA does not appl
 }}
 
-By looking for keywords such as 'privacy policy', 'cookie policy', and more, in a <a hreflang="en" href="https://github.com/RUB-SysSec/we-value-your-privacy/blob/master/privacy_wording.json">number of languages</a>, we see that almost 40% of websites refer to some sort of privacy policy. While some websites are not required to have such a policy, many websites handle personal data and should ther ...
+By looking for keywords such as "privacy policy", "cookie policy", and more, in a <a hreflang="en" href="https://github.com/RUB-SysSec/we-value-your-privacy/blob/master/privacy_wording.json">number of languages</a>, we see that almost 39.70% of mobiles websites, and 43.02% of desktop sites refer to some sort of privacy policy. While some websites are not required to have such a policy, many websi ...
```
```suggestion
By looking for keywords such as "privacy policy", "cookie policy", and more, in a <a hreflang="en" href="https://github.com/RUB-SysSec/we-value-your-privacy/blob/master/privacy_wording.json">number of languages</a>, we see that almost 39.70% of mobile websites, and 43.02% of desktop sites refer to some sort of privacy policy. While some websites are not required to have such a policy, many websites handle personal data and should therefore have a privacy policy to be fully transparent towards their users.
```

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -579,51 +551,44 @@ Since FLoC was an experiment, it was not widely deployed. Instead, websites coul
+  description="Bar chart showing the percentage of cookies with the SameParty cookie attribute according to the request context. For first-party cookies `SameParty` is used on 38 desktop sites and 73 mobile sites for third-party cookies it's 2,527 for desktop sites and 1,805 mobile sites.",
   chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRfOwcNkLp-mYBkmhDV5AFxl8p0ls9oxFnmmo0WUcAJxjxQqmHjquRZCWj_lNZRyFtX3RdH5T92IESu/pubchart?oid=935824621&format=interactive",
   sheets_gid="858972835",
   sql_file=""
```
Yes. Use relative paths. Is this the right SQL file?

```suggestion
  sql_file="../security/cookie_attributes.sql"
```

tunetheweb commented on src/content/en/2021/privacy.md:
```diff
@@ -5,572 +5,544 @@ description: Privacy chapter of the 2021 Web Almanac covering adoption and impac
+## Introduction
 
+[_"On the Internet, nobody knows you're a dog."_](https://en.wikipedia.org/wiki/On_the_Internet,_nobody_knows_you%27re_a_dog) While it might be true that you could try to remain anonymous to use the Internet as such, it can be quite hard to keep your personal data fully private. A [whole industry](https://crackedlabs.org/en/corporate-surveillance/) is dedicated to tracking users online, to build  ...
 
```
Isn't the whole chapter about that? 😁

Here's the original: https://20211109t172922-dot-webalmanac.uk.r.appspot.com/en/2021/privacy?print
(I've used the `?print` query param to show the original images, since I changed some of the graphs)

And the new version: https://20211112t221945-dot-webalmanac.uk.r.appspot.com/en/2021/privacy

I think the "breathing space" makes it easier to read. Especially on mobile. But OK to merge some of these back in if you feel strongly about this. Do you want to merge the ones you feel should be re-merged?

VictorLeP replied:
Everywhere else, the breathing space is an improvement indeed, thanks! Just in this one spot I find that it breaks the flow :) We could break just before "A whole industry", but then the first paragraph might become too short?

tunetheweb replied:
SGTM

tunetheweb commented:
Redployed to the same staging link: https://20211112t221945-dot-webalmanac.uk.r.appspot.com/en/2021/privacy with @VictorLeP 's edits.

tunetheweb commented:
@ydimova I can't query `bamboo-chariot-267911.cname_analysis.results_tracking_2021_08_01` as get a permission denied error.

I'd like to convert these to percentages and add mobile for consistency but cannot do this if I cannot access the data.

It is also using the August data - do we have the same for July?

Also where does this data come from?

tunetheweb commented:
@ydimova as discussed offline, I've added the CNAME data as a table in the `httparchive.almanac` schema.

I've also updated the SQL and rerun to make them percentage based, and based on number of pages, rather than registered domain. This does change some of the numbers (changes shown in this commit: https://github.com/HTTPArchive/almanac.httparchive.org/pull/2475/commits/67e6defb722ce05fcd77bf4f1c57d5ae60cdf0b9), but don't think it changes the point of the text.

I've restaged it to the same URL: https://20211112t221945-dot-webalmanac.uk.r.appspot.com/en/2021/privacy

I'm going to go ahead and merge this so it can be included in the early release in the next day or two, until the assumption you won't have an issue with this. Feel free to open a new PR if I got anything wrong.

Thanks again for your efforts on this and for responding quickly to this feedback. Great to get the chapter done!
